compiler/noirc_frontend/src/hir/type_check/errors.rs

```diff
@@ -369,11 +369,37 @@ impl<'a> From<&'a TypeCheckError> for Diagnostic {
                 )
             }
             TypeCheckError::TypeKindMismatch { expected_kind, expr_kind, expr_location } => {
-                Diagnostic::simple_error(
-                    format!("Expected kind {expected_kind}, found kind {expr_kind}"),
-                    String::new(),
-                    *expr_location,
-                )
+                // Try to improve the error message for some kind combinations
+                match (expected_kind, expr_kind) {
+                    (Kind::Normal, Kind::Numeric(_)) => {
+                        Diagnostic::simple_error(
+                            "Expected type, found let parameter".into(),
```

Comment:
I think this may be less confusing if we use the term "numeric generic" since that is what the docs use. I think the cases under this should use that term as well instead of referring to them as constants, which they may not be.

Reply:
Thanks! I copied Rust in most cases but I wasn't sure about that name either. I'll change it.

Reply:
Done!